vpr/src/draw/intra_logic_block.cpp

```diff
@@ -434,19 +434,22 @@ static void draw_internal_pb(const ClusterBlockId clb_index, t_pb* pb, const ezg
         g->draw_rectangle(abs_bbox);
     }
 
-    // draw text for each physical block.
-    // format: <block_type_name>:<mode_name>[<placement_index>]
+    // Display text for each physical block.
     std::string pb_type_name(pb_type->name);
 
-    // get the mode of the physical block
     if (!pb->is_primitive()) {
-        // primitives have no modes
+        // Format for non-primitives: <block_type_name>[<placement_index>]:<mode_name>
+        pb_type_name += "[" + std::to_string(pb->pb_graph_node->placement_index) + "]";
```

Comment:
Can we also not print the mode if the mode_name == pb_type_name (i.e. it provides redundant info)?  Would require that if statement.

Reply:
Looks like you just did that in the next commit :).